### PR TITLE
Drop support for Symfony 4

### PR DESCRIPTION
**Pull request by greg0ire**
It is unmaintained and makes contributions unnecessarily hard to do.

### CODE REVIEW

**Comment by greg0ire** [comment]
Ah… PHP 7.1, of course.

**Comment by derrabus** [comment]
You can't do that without dropping PHP 7.1. 🤷🏻

**Comment by greg0ire** [comment]
Maybe we should ?

**Comment by derrabus** [comment]
These are the packagist statistics for the 2.16 branch (2.17 is too new, 2.15 looks similar).

Bildschirmfoto 2023-11-17 um 09 01 50

So, people who do upgrade their ORM, don't run PHP 7.1 anymore. And we don't need to care about people who don't upgrade their ORM anyway, I guess.

Let's bump to 7.4 and kick out DBAL 2 while we're at it?

Then again, all of this is what we've planned for ORM 3 anyway. Maybe we just flag issues related to Symfony 4 as won't fix and try to push ORM 3 out of the door instead?

**Comment by greg0ire** [comment]
> Maybe we just flag issues related to Symfony 4 as won't fix

As you can see, it's more involving than setting a label on some issues. Sometimes, we have to go out of our way to make sure a test does not run when using Symfony 4. And I think the same happens for PHP 7.1.

**Comment by derrabus** [comment]
I know. I can live with both solutions. Pick one. 🙂

**Comment by greg0ire** [comment]
> Let's bump to 7.4 and kick out DBAL 2 while we're at it?

I'd say let's try to notice when supporting an unmaintained version becomes a hindrance, and drop exactly what causes the hindrance, otherwise @beberlei is going to send counter-PRs because stuff actually still works with PHP 7.2, for instance. So to me the balance we need to find is between "it works" and "it makes contributions and maintenance" painful.

**Comment by derrabus** [comment]
> I'd say let's try to notice when supporting an unmaintained version becomes a hindrance

But that's not really the case here. The only real Symfony dependencies we have are Console and maybe YAML and Cache. Neither of them causes trouble.

If you're referring to the serialization issues with VarExporter 4.4: We don't use anything from VarExporter 4.4. And we don't tell anyone to even use VarExporter to serialize our stuff. If we merge any compatibility fixes for VarExporter 4.4, it's because we're being nice and not because we're identifying the incompatibility to a EOL library that we don't even depend on as a bug. Dropping support for Console 4.4 (which is _effectively_ what your PR does) would not change that policy for me.

**Comment by greg0ire** [comment]
I was indeed referring to the serialization issues with VarExporter

> We don't use anything from VarExporter 4.4

:thinking: we use the lazy ghosts, do we not? We do depend on that lib, it's just an optional dependency, and it becomes a regular dependency on 3.0.x

I'm dropping support for Symfony 4 because in general, I think our policy is to drop a version of Symfony accross all Symfony Components.

**Comment by derrabus** [comment]
> > We don't use anything from VarExporter 4.4
> 
> 🤔 we use the lazy ghosts, do we not?

Yes, from VarExporter >= 6.2. VarExporter 4.4 does not have lazy ghosts. It's a neutral package for us. We don't actively support it because the is nothing to support.

This PR drops support for Console 4.4 (which does not cause us trouble) and does not even prevent people from using VarExporter 4.4 on ORM classes. It fixes no problem that we actually have.

I see two options:

* Get ORM 3 out of the door soon and leave ORM 2 as it is. Feature development will continue on 3.1.x with shiny new dependencies.
* Do dependency housekeeping on ORM 2: PHP >= 7.4, DBAL >= 3.7, Symfony >= 5.4. Given that we will maintain ORM 2 for a while after 3.0.0, this might be a reasonable thing to do anyway.

**Comment by greg0ire** [comment]
Ok I see why the 4.4 was important in "VarExporter 4.4" now.

> It fixes no problem that we actually have.

I'd argue we have needlessly complex tests and contributions, and that it fixes that. So, no end user problem fixed, but I think we should take contributors and maintainers into account.

> Do dependency housekeeping on ORM 2: PHP >= 7.4, DBAL >= 3.7, Symfony >= 5.4. Given that we will maintain ORM 2 for a while after 3.0.0, this might be a reasonable thing to do anyway.

If @beberlei signs off on this I'm more than happy to go with that solution.

File: composer.json [hunk 1]
```diff
@@ -34,7 +34,7 @@
         "doctrine/lexer": "^2",
         "doctrine/persistence": "^2.4 || ^3",
         "psr/cache": "^1 || ^2 || ^3",
-        "symfony/console": "^4.2 || ^5.0 || ^6.0 || ^7.0",
+        "symfony/console": "^5.0 || ^6.0 || ^7.0",
```
Comment (Chris53897) [inline composer.json:37]: I guess this line (and for symfony/yaml) can be changed to `^5.4 || ^6.0 || ^7.0` to only support LTS Versions?

Comment (greg0ire) [reply composer.json:37]: yeah I guess

**Comment by greg0ire** [comment]
Clear no from Benjamin.

**Pull request closed without merging**